Issue #374: Fix SingleBreakOrContinue violations

yaziza opened the pull request.

yaziza commented:
Update: 6/7 violations resolved.

Im still working on the last violation. After fixing i will extend the unit tests to meet the requirements.

yaziza commented on sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/CustomDeclarationOrderCheck.java:
```diff
@@ -596,37 +596,27 @@ private int getPositionInOrderDeclaration(final DetailAST ast)
+                            || classDetail.containsSetter(methodName)
+                            || isMainMethod(ast)) {
                         result = index;
                         break;
```
After reducing the number of breaks to 2, not sure how to eleminate the last one ...
Maybe completly refactoring this method ...

romani replied:
code is not simple :), this is just good example that need to follow new Checks ourself.

solution is to put `result != null` condition to FOR, and remove `break` completely.

yaziza replied:
Sorry, i didn't get it. 

What about setting: index = Integer.MAX_VALUE - 1; instead f break ?

romani replied:
make loop like `for (int index = 0; index < customOrderDeclaration.size() && result != 1 ; index++) `
no breaks are required.

romani commented:
please remove from you commit all unrelated changes(that are just formatting), or move them in separate commit of this PR. This will ease diff verification.

romani commented:
Travis is not happy with reduced coverage
> [ERROR] Jan 11, 2016 7:59:02 PM net.sourceforge.cobertura.coveragedata.CoverageDataFileHandler loadCoverageData
INFO: Cobertura: Loaded information on 63 classes.
com.github.sevntu.checkstyle.checks.coding.DiamondOperatorForVariableDefinitionCheck failed check. Branch coverage rate of 85.0% is below 90.0%

It might be resulted by removing the code lines. Please generate coverage report and share pictures of coverage before and after for this method. If it is the same it is ok to change configuration(see cobertura section `pom.xml`) there limit is specified.

romani commented:
@yaziza , 1.18 release will be this Sunday, please find time to finish this PR.

yaziza commented:
Fixed, i have added a new case in the test input file and the branch cover is now 94%
Also i have removed the unnesseray != null check, which will be checked anyway in equals.

Thank you for your support !

Do you have any suggestions on issues i can tackle next ?

romani commented:
merged as FF. Thanks a lot for your time and contributions.

> Also i have removed the unnesseray != null check

One of the banefit of 100%  coverage is to remove dead/extra code. 

> Do you have any suggestions on issues i can tackle next ?

Can you help us with #410 ? Exceptions during execution usually force user to deactivate the Check at all. But you can choose any unassigned issue.
